src/view/event-edit-view.js

```diff
@@ -218,60 +213,132 @@ const createEventEditTemplate = (
+  #allOffers = null;
 
   #handleFormSubmit = null;
   #handleFormClose = null;
 
   constructor({
     event = NEW_EVENT,
-    eventInfo = NEW_EVENT_INFO,
     allDestinations,
-    availableOffers,
+    allOffers,
     onFormSubmit,
     onFormClose,
   }) {
     super();
-    this.#event = event;
-    this.#eventInfo = eventInfo;
+    this._setState(EventEditView.parseEventToState(event));
     this.#allDestinations = allDestinations;
-    this.#availableOffers = availableOffers;
+    this.#allOffers = allOffers;
     this.#handleFormSubmit = onFormSubmit;
     this.#handleFormClose = onFormClose;
-
-    this.form = this.element.querySelector('form');
-    this.form.addEventListener('submit', this.#formSubmitHandler);
-    this.element
-      .querySelector('.event__rollup-btn')
-      .addEventListener('click', this.#formCloseHandler);
+    this._restoreHandlers();
   }
 
   get template() {
     return createEventEditTemplate(
-      this.#event,
-      this.#eventInfo,
+      this._state,
       this.#allDestinations,
-      this.#availableOffers
+      this.#allOffers
     );
   }
 
-  resetForm = () => {
-    this.form.reset();
+  _restoreHandlers() {
+    this.element
+      .querySelector('form')
+      .addEventListener('submit', this.#formSubmitHandler);
+    this.element
+      .querySelector('.event__rollup-btn')
+      .addEventListener('click', this.#formCloseHandler);
+    this.element
+      .querySelector('.event__type-group')
+      .addEventListener('change', this.#typeChangeHandler);
+    this.element
+      .querySelector('.event__input--destination')
+      .addEventListener('change', this.#destinationChangeHandler);
+    this.element
+      .querySelector('.event__field-group--price')
+      .addEventListener('change', this.#priceChangeHandler);
+
+    if (this.element.querySelector('.event__section--offers') !== null) {
+      this.element
+        .querySelector('.event__section--offers')
+        .addEventListener('change', this.#offerChangeHandler);
+    }
+  }
+
+  reset = (event) => {
+    this.updateElement(EventEditView.parseEventToState(event));
   };
 
   #formSubmitHandler = (evt) => {
     evt.preventDefault();
-    this.#handleFormSubmit(this.#event);
+    this.#handleFormSubmit(EventEditView.parseStateToEvent(this._state));
   };
 
   #formCloseHandler = (evt) => {
     evt.preventDefault();
-    //this.form.reset();
-    this.resetForm();
     this.#handleFormClose();
   };
+
+  #typeChangeHandler = (evt) => {
+    evt.preventDefault();
+    if (evt.target.tagName === 'INPUT') {
+      this.updateElement({
+        type: evt.target.value,
+        offers: [],
+      });
+    }
+  };
+
+  #destinationChangeHandler = (evt) => {
+    const name = evt.target.value;
+    const newDestination = this.#allDestinations.find(
+      (destination) => destination.name === name
+    );
+
+    if (newDestination) {
+      this.updateElement({
+        destination: newDestination.id,
+      });
+    }
```

Comment:
Можно сделать обратный `if` и немного снизить вложенность. Но это вкусовщина, конечно.

```suggestion
    if (!newDestination) {
      return
    }
      
    this.updateElement({
        destination: newDestination.id,
    });
```

Reply:
Ок, принято